File: backend_dart_kpis/lib/controllers/formulario_controller.dart
import 'dart:convert';
import 'package:shelf/shelf.dart';
import '../services/formulario_service.dart';
import '../sockets/realtime_gateway.dart';

Future<Response> crearRespuesta(Request req) async {
  try {
    final body = await req.readAsString();
    final data = jsonDecode(body) as Map<String, dynamic>;
    final input = FormularioInput.fromMap(data);

    await insertarRespuesta(input);
    await emitirKPIsInstantaneo();

    return Response.ok(jsonEncode({'ok': true}),
        headers: {'Content-Type': 'application/json'});
  } catch (e) {
    return Response.internalServerError(
      body: jsonEncode({'error': 'No se pudo guardar la respuesta', 'detalle': e.toString()}),
      headers: {'Content-Type': 'application/json'},
    );
  }
}

File: backend/src/controllers/formulario.controller.js
import { insertarRespuesta } from '../services/formulario.service.js';
import { emitirKPIsInstantaneo } from '../sockets/realtime.gateway.js';

export async function crearRespuesta(req, res) {
  try {
    const payload = req.body; 
    await insertarRespuesta(payload);

    await emitirKPIsInstantaneo(req.app.get('io') || global.__io);

    res.json({ ok: true });
  } catch (err) {
    console.error(err);
    res.status(500).json({ error: 'No se pudo guardar la respuesta' });
  }
}
